src/test/java/com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinterTest.java

```diff
@@ -100,12 +95,79 @@ public void testCreationOfFakeCommentBlock() throws Exception {
 
     @Test
     public void testNoUnnecessaryTextinJavadocAst() throws Exception {
-        final String actual = DetailNodeTreeStringPrinter.printFileAst(
-                new File(getPath("InputNoUnnecessaryTextInJavadocAst.javadoc")))
-                .replaceAll("\\\\r\\\\n", "\\\\n");
-        final String expected = new String(Files.readAllBytes(Paths.get(
-                getPath("expectedNoUnnecessaryTextInJavadocAst.txt"))), StandardCharsets.UTF_8)
-                .replaceAll("\\\\r\\\\n", "\\\\n");
-        assertEquals(expected, actual);
+        verifyJavadocTree(getPath("expectedNoUnnecessaryTextInJavadocAst.txt"),
+                getPath("InputNoUnnecessaryTextInJavadocAst.javadoc"));
+    }
+
+    @Test
+    public void testUnescapedJavaCodeWithGenericsInJavadoc() throws IOException {
+        try {
+            DetailNodeTreeStringPrinter.printFileAst(new File(getPath(
+                    "InputUnescapedJavaCodeWithGenericsInJavadoc.javadoc")));
+        }
+        catch (IllegalArgumentException ex) {
+            final String expected = getParseErrorMessage(new ParseErrorMessage(
+                    35, MSG_JAVADOC_MISSED_HTML_CLOSE, 7, "parsing"));
+            assertEquals(expected, ex.getMessage());
+        }
+    }
+
+    @Test
+    public void testNoViableAltException() throws IOException {
+        LocalizedMessage.setLocale(Locale.ROOT);
```

Comment:
we should not use `Locale.ROOT` , it might be still required somewhere but that should be explained in javadoc. I doubt we can not use localized test here.

> hrough a check?

yes, but we still did not finish transition from non localized tests - #3110 , it might force us to create a lot of suppressions.